src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -1067,86 +1073,363 @@ def inner_product_matrix(self):
+
+            sage: QuaternionAlgebra(RR(2.),1).ramified_primes()
+            Traceback (most recent call last):
+            ...
+            ValueError: base field must be rational numbers or a number field
         """
-        if not is_RationalField(self.base_ring()):
-            raise ValueError("base field must be the rational numbers")
+        return self.ramified_places(inf=False)
+
+    @cached_method
+    def discriminant(self):
+        r"""
+        Return the discriminant of this quaternion algebra.
+
+        The discriminant of a quaternion algebra over a number field is the
+        product of the finite places at which the algebra ramifies.
+
+        OUTPUT:
+
+        The discriminant of this quaternion algebra, given as
+
+        - an element of `\ZZ` if the algebra is defined over `\QQ`,
+
+        - an integral fractional ideal of the base number field, otherwise.
+
+        EXAMPLES::
+
+            sage: QuaternionAlgebra(210, -22).discriminant()
+            210
+            sage: QuaternionAlgebra(19).discriminant()
+            19
+            sage: QuaternionAlgebra(-1, -1).discriminant()
+            2
+
+        Some examples over number fields::
 
-        return sorted([p for p in set([2]).union(prime_divisors(self._a.numerator()),
-                prime_divisors(self._a.denominator()), prime_divisors(self._b.numerator()),
-                prime_divisors(self._b.denominator())) if hilbert_symbol(self._a, self._b, p) == -1])
+            sage: K = QuadraticField(3)
+            sage: L = QuadraticField(-15)
+            sage: QuaternionAlgebra(K, -1, -1).discriminant()
+            Fractional ideal (1)
+            sage: QuaternionAlgebra(L, -1, -1).discriminant()
+            Fractional ideal (2)
+
+        We can also use number field elements as invariants::
+
+            sage: x = polygen(ZZ, 'x')
+            sage: F.<a> = NumberField(x^2 - x - 1)
+            sage: QuaternionAlgebra(F, 2*a, F(-1)).discriminant()
+            Fractional ideal (2)
+
+        The method does not make sense over an arbitrary base ring::
+
+            sage: QuaternionAlgebra(RR(2.),1).discriminant()
+            Traceback (most recent call last):
+            ...
+            ValueError: base field must be rational numbers or a number field
+        """
+        F = self.base_ring()
+        if is_RationalField(F):
+            return ZZ.prod(self.ramified_places(inf=False))
+
+        return F.ideal(F.prod(self.ramified_places(inf=False)))
 
     def is_isomorphic(self, A) -> bool:
         """
-        Return ``True`` if (and only if) ``self`` and ``A`` are isomorphic quaternion algebras over Q.
+        Check whether this quaternion algebra is isomorphic to ``A``.
+
+        Currently only implemented for quaternion algebras defined over
+        a number field; based on Main Theorem 14.6.1 in [Voi2021]_,
+        noting that `\\QQ` has a unique infinite place.
```

Comment:
better to change the docstring to raw and avoid the double backslash

Reply:
Alright, I'll change it - is there a reason why one is preferable over the other one, though? The developer guide gives both as valid options; maybe that preference is worth stating there.

Reply:
See #37508